traitlets/traitlets.py

```diff
@@ -1910,8 +1910,14 @@ def info(self):
 
 class CaselessStrEnum(Enum):
     """An enum of strings where the case should be ignored."""
-
+    
+    def __init__(self, values, default_value=Undefined, **metadata):
+        values = [py3compat.cast_unicode_py2(value) for value in values]
+        super(CaselessStrEnum, self).__init__(values, default_value=default_value, **metadata)
+    
     def validate(self, obj, value):
+        if not py3compat.PY3 and isinstance(value, str):
+            value = value.decode('utf8')
```

Comment:
Yes - me being inconsistent about whether or not to use py3compat from one minute to the next. Should be more consistent now. It does leave the check for `str` in case of allow_none.